nixos/modules/services/web-apps/youtrack.nix

```diff
@@ -1,131 +1,178 @@
-    };
-
-    package = mkPackageOption pkgs "youtrack" { };
-
-    port = mkOption {
+    port = lib.mkOption {
       description = lib.mdDoc ''
         The port youtrack will listen on.
       '';
       default = 8080;
-      type = types.port;
+      type = lib.types.port;
     };
 
-    statePath = mkOption {
+    statePath = lib.mkOption {
       description = lib.mdDoc ''
-        Where to keep the youtrack database.
+        Path were the YouTrack state is stored.
+        To this path the base version (e.g. 2023_1) of the used package will be appended.
       '';
-      type = types.path;
+      type = lib.types.path;
       default = "/var/lib/youtrack";
     };
 
-    virtualHost = mkOption {
+    virtualHost = lib.mkOption {
       description = lib.mdDoc ''
         Name of the nginx virtual host to use and setup.
         If null, do not setup anything.
       '';
       default = null;
-      type = types.nullOr types.str;
+      type = lib.types.nullOr lib.types.str;
     };
 
-    jvmOpts = mkOption {
+    jvmOpts = lib.mkOption {
       description = lib.mdDoc ''
         Extra options to pass to the JVM.
         See https://www.jetbrains.com/help/youtrack/standalone/Configure-JVM-Options.html
         for more information.
       '';
-      type = types.separatedString " ";
-      example = "-XX:MetaspaceSize=250m";
+
+      example = "--J-XX:MetaspaceSize=250m";
       default = "";
     };
 
-    maxMemory = mkOption {
+    maxMemory = lib.mkOption {
       description = lib.mdDoc ''
         Maximum Java heap size
       '';
-      type = types.str;
+      type = lib.types.str;
       default = "1g";
     };
 
-    maxMetaspaceSize = mkOption {
+    maxMetaspaceSize = lib.mkOption {
       description = lib.mdDoc ''
         Maximum java Metaspace memory.
       '';
-      type = types.str;
+      type = lib.types.str;
       default = "350m";
     };
-  };
 
-  config = mkIf cfg.enable {
-
-    systemd.services.youtrack = {
-      environment.HOME = cfg.statePath;
-      environment.YOUTRACK_JVM_OPTS = "${extraAttr}";
-      after = [ "network.target" ];
-      wantedBy = [ "multi-user.target" ];
-      path = with pkgs; [ unixtools.hostname ];
-      serviceConfig = {
-        Type = "simple";
-        User = "youtrack";
-        Group = "youtrack";
-        Restart = "on-failure";
-        ExecStart = ''${cfg.package}/bin/youtrack --J-Xmx${cfg.maxMemory} --J-XX:MaxMetaspaceSize=${cfg.maxMetaspaceSize} ${cfg.jvmOpts} ${cfg.address}:${toString cfg.port}'';
+    settings = lib.mkOption {
+      type = with lib.types; attrsOf (oneOf [ bool int str ]);
+      description = lib.mdDoc ''
+        Settings directly set in a YouTrack configuration file.
+        See https://www.jetbrains.com/help/youtrack/server/2023.1/YouTrack-Java-Start-Parameters.html
+        for more information.
+      '';
+      example = {
+        tls-redirect-from-http = true;
+        jetbrains.youtrack.admin.restore = true;
```

Comment:
Looks like options from [general parameter](https://www.jetbrains.com/help/youtrack/server/2023.1/YouTrack-Java-Start-Parameters.html#general-parameters) need to be specified as JVM options, i.e. something like `youtrack configure -J-Djetbrains.youtrack.admin.restore=true`, so this particular example doesn't look correct.
Specifying [environment parameters](https://www.jetbrains.com/help/youtrack/server/2023.1/YouTrack-Java-Start-Parameters.html#environmental-parameters) works just fine :relieved: